pymc/backends/arviz.py

```diff
@@ -396,34 +376,35 @@ def log_likelihood_to_xarray(self):
             ),
         )
 
-    def translate_posterior_predictive_dict_to_xarray(self, dct) -> xr.Dataset:
+    def translate_posterior_predictive_dict_to_xarray(self, dct, kind) -> xr.Dataset:
         """Take Dict of variables to numpy ndarrays (samples) and translate into dataset."""
         data = {}
+        warning_vars = []
         for k, ary in dct.items():
-            shape = ary.shape
-            if shape[0] == self.nchains and shape[1] == self.ndraws:
+            if (ary.shape[0] == self.nchains) and (ary.shape[1] == self.ndraws):
                 data[k] = ary
-            elif shape[0] == self.nchains * self.ndraws:
-                data[k] = ary.reshape((self.nchains, self.ndraws, *shape[1:]))
             else:
                 data[k] = np.expand_dims(ary, 0)
-                # pylint: disable=line-too-long
-                _log.warning(
-                    "posterior predictive variable %s's shape not compatible with number of chains and draws. "
-                    "This can mean that some draws or even whole chains are not represented.",
-                    k,
-                )
+                warning_vars.append(k)
+        warnings.warn(
```

Comment:
@OriolAbril I think you intended for this warning to only be printed if `warnings_vars` was not empty. Is this correct? I am currently always getting the output:
```python
/Users/thomas/repo/pymc/pymc/backends/arviz.py:389: UserWarning: The shape of variables  in posterior_predictive group is not compatible with number of chains and draws. The automatic dimension naming might not have worked. This can also mean that some draws or even whole chains are not represented.
```

Reply:
To be extra clear: The warning prints a space in place of `{', '.join(warning_vars)}`.

Reply:
Yeah, sorry I also missed it in my review.
It needs to be indented into an `if warning_vars`.
Thanks for reporting this! Do you want to open a PR? Otherwise I'll do it in the morning